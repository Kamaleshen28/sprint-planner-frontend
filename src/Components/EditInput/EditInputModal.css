.story-container-edit {
  width: 90%;
}
