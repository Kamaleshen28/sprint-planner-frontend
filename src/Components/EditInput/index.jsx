import React, { useState } from 'react';
import {
  Box,
  Chip,
  Checkbox,
  FormControl,
  Fab,
  InputLabel,
  MenuItem,
  Select,
  OutlinedInput,
  ListItemText,
} from '@mui/material';
import PropTypes from 'prop-types';
import { ValidatorForm, TextValidator } from 'react-material-ui-form-validator';
import './EditInputModal.css';

const ITEM_HEIGHT = 48;
const ITEM_PADDING_TOP = 8;
const MenuProps = {
  PaperProps: {
    style: {
      maxHeight: ITEM_HEIGHT * 4.5 + ITEM_PADDING_TOP,
      width: 250,
    },
  },
};

Item.propTypes = {
  sx: PropTypes.oneOfType([
    PropTypes.arrayOf(
      PropTypes.oneOfType([PropTypes.func, PropTypes.object, PropTypes.bool]),
    ),
    PropTypes.func,
    PropTypes.object,
  ]),
};

function Item(props) {
  const { sx, ...other } = props;
  return (
    <Box
      sx={{
        p: 1,
        m: 1,
        width: 200,
        // bgcolor: (theme) =>
        //   theme.palette.mode === 'dark' ? '#101010' : 'grey.100',
        color: (theme) =>
          theme.palette.mode === 'dark' ? 'grey.300' : 'grey.800',
        // border: '1px solid',
        // borderColor: (theme) =>
        //   theme.palette.mode === 'dark' ? 'grey.800' : 'grey.300',
        // borderRadius: 2,
        fontSize: '0.875rem',
        fontWeight: '700',
        ...sx,
      }}
      {...other}
    />
  );
}

const EditInput = ({
  id,
  stories,
  dependencies,
  developer,
  storyPoints,
  storyList,
  setStoryList,
  developerList,
  closeedit,
}) => {
  const [editstories, seteditStories] = useState(stories);
  const [editdependencies, seteditDependencies] = useState(dependencies);
  const [editdeveloper, seteditDeveloper] = useState(developer);
  const [editstoryPoints, seteditStoryPoints] = useState(storyPoints);
  const handleDeveloperChangeEdit = (event) => {
    const { value } = event.target;
    if (value === ' ') {
      seteditDeveloper([]);
    } else {
      seteditDeveloper([value]);
    }
  };
  const handledependencyChangeEdit = (event) => {
    const { value } = event.target;
    seteditDependencies(value);
  };
  const handleSubmitEdit = (e) => {
    console.log('edit');
    console.log(
      id,
      editstories,
      editdependencies,
      editdeveloper,
      editstoryPoints,
    );
    e.preventDefault();
    if (editstories && editstoryPoints) {
      const newStory = {
        id: id,
        stories: editstories,
        dependencies: editdependencies,
        developer: editdeveloper,
        storyPoints: editstoryPoints,
      };
      const index = storyList.findIndex((story) => story.id === id);
      const newStoryList = [...storyList];
      newStoryList[index] = newStory;
      setStoryList(newStoryList);
      closeedit();
    } else {
      console.log('error');
    }
  };
  return (
    <ValidatorForm
      className="story-container-edit"
      onSubmit={handleSubmitEdit}
      // onError={(errors) => console.log(errors)}
    >
      <Box
        sx={{
          display: 'grid',
          gridTemplateColumns: '1fr repeat(4, 2fr) 1fr 1fr',
          height: '100px',
          width: '90%',
          alignItems: 'center',
          bgcolor: 'white',
        }}
      >
        <div></div>
        <Item sx={{ width: '80%', mx: 'auto' }}>
          <TextValidator
            style={{ width: '100%' }}
            type="text"
            name="story"
            defaultValue="Disabled"
            value={editstories}
            placeholder="Enter Story Title"
            validators={['required']}
            errorMessages={['this field is required']}
            onChange={(e) => seteditStories(e.target.value)}
          />
        </Item>
        <FormControl
          sx={{ m: 1, width: 300 }}
          style={{ backgroundColor: 'white', borderRadius: '10px' }}
        >
          <InputLabel
            variant="standard"
            htmlFor="uncontrolled-native"
            sx={{ m: 1 }}
          >
            Story Dependecy
          </InputLabel>
          <Select
            sx={{ m: 1 }}
            labelId="demo-multiple-name-label"
            id="demo-multiple-name"
            multiple
            value={editdependencies}
            onChange={handledependencyChangeEdit}
            input={<OutlinedInput id="select-multiple-chip" label="Chip" />}
            renderValue={(selected) => (
              <Box sx={{ display: 'flex', flexWrap: 'wrap', gap: 0.5 }}>
                {selected.map((dependencyId) => (
                  <Chip
                    key={dependencyId}
                    label={
                      storyList?.find((e) => e.id === dependencyId).stories
                    }
                  />
                ))}
              </Box>
            )}
            MenuProps={MenuProps}
          >
            {storyList.map((story) => {
              return (
                story.id !== id && (
                  <MenuItem key={story.id} value={story.id}>
                    <Checkbox checked={editdependencies?.includes(story.id)} />
                    <ListItemText primary={story.stories} />
                  </MenuItem>
                )
              );
            })}
          </Select>
        </FormControl>
        <FormControl
          sx={{ m: 1, width: 300 }}
          style={{ backgroundColor: 'white', borderRadius: '10px' }}
        >
          <InputLabel
            variant="standard"
            htmlFor="uncontrolled-native"
            sx={{ m: 1 }}
          >
            Preassign Developer
          </InputLabel>
          <Select
            sx={{ m: 1 }}
            value={editdeveloper}
            onChange={handleDeveloperChangeEdit}
            input={<OutlinedInput />}
            renderValue={(selected) => {
              const obj = developerList?.find((e) => e.id === selected[0]);
              return obj?.developer;
            }}
            MenuProps={MenuProps}
          >
            <MenuItem value=" ">
              <em>None</em>
            </MenuItem>
            {developerList.map((developer) => (
              <MenuItem key={developer.id} value={developer.id}>
                {developer.developer}
              </MenuItem>
            ))}
          </Select>
        </FormControl>
        <Item sx={{ width: '80%', mx: 'auto' }}>
          <TextValidator
            style={{ width: '100%' }}
            type="text"
            name="storyPoints"
            value={editstoryPoints}
            placeholder="Enter Story Points"
            validators={['required', 'isNumber', 'minNumber:0', 'maxNumber:10']}
            errorMessages={[
              'this field is required',
              'It should be number',
              'Number should be positive',
            ]}
            onChange={(e) => seteditStoryPoints(e.target.value)}
          />
        </Item>
        <Fab
          // variant="contained"
          color="primary"
          type="submit"
          aria-label="add"
          sx={{ fontSize: '1.5rem' }}
        >
          +
        </Fab>
        <Fab
          onClick={closeedit}
          variant="contained"
          style={{ margin: 'auto', display: 'flex' }}
          color="error"
          type="submit"
        >
          x
        </Fab>
      </Box>
    </ValidatorForm>
  );
};

EditInput.propTypes = {
  id: PropTypes.number,
  stories: PropTypes.string,
  setStories: PropTypes.func,
  dependencies: PropTypes.array,
  setDependencies: PropTypes.func,
  developer: PropTypes.array,
  setDeveloper: PropTypes.func,
  storyPoints: PropTypes.string,
  setStoryPoints: PropTypes.func,
  storyList: PropTypes.array,
  setStoryList: PropTypes.func,
  developerList: PropTypes.array,
  closeedit: PropTypes.func,
};

export default EditInput;
