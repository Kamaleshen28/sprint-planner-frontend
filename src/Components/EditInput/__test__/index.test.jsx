import React from 'react';
import { render, screen } from '@testing-library/react';

import EditInput from '..';

describe('EditInput', () => {
  it('should render correctly', () => {
    const mockProps = {
      id: 1,
      strories: 'test',
      setStories: jest.fn(),
      dependencies: [],
      setDependencies: jest.fn(),
      developer: [],
      setDeveloper: jest.fn(),
      storyPoints: '1',
      setStoryPoints: jest.fn(),
      storyList: [],
      setStoryList: jest.fn(),
      developerList: [],
      closeedit: jest.fn(),
    };
    const { asFragment } = render(<EditInput {...mockProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
