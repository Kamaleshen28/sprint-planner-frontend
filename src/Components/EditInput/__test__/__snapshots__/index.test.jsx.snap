// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditInput should render correctly 1`] = `
<DocumentFragment>
  <form
    class="story-container-form"
  >
    <div
      class="MuiBox-root css-1lxsugo"
    >
      <div />
      <div
        class="MuiBox-root css-up619a"
      >
        <div>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r0:"
                name="story"
                placeholder="Enter Story Title"
                type="text"
                value="Disabled"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiFormControl-root css-2m9kme-MuiFormControl-root"
        style="background-color: white; border-radius: 10px;"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-1akq2c8-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="uncontrolled-native"
        >
          Story Dependecy
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-4u1c9l-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="demo-multiple-name-label demo-multiple-name"
            class="MuiSelect-select MuiSelect-outlined MuiSelect-multiple MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="demo-multiple-name"
            role="button"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            id="select-multiple-chip"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-yjsfm1"
            >
              <span>
                Chip
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiFormControl-root css-2m9kme-MuiFormControl-root"
        style="background-color: white; border-radius: 10px;"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-standard css-1akq2c8-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="uncontrolled-native"
        >
          Preassign Developer
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-4u1c9l-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            role="button"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiBox-root css-up619a"
      >
        <div>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r1:"
                name="storyPoints"
                placeholder="Enter Story Points"
                type="text"
                value="1"
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiFab-root MuiFab-contained MuiFab-sizeLarge MuiFab-primary MuiFab-root MuiFab-contained MuiFab-sizeLarge MuiFab-primary css-1fo2sjy-MuiButtonBase-root-MuiFab-root"
        style="margin: auto; display: flex;"
        tabindex="0"
        type="submit"
      >
        +
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiFab-root MuiFab-contained MuiFab-sizeLarge MuiFab-secondary MuiFab-root MuiFab-contained MuiFab-sizeLarge MuiFab-secondary css-1p7a5wk-MuiButtonBase-root-MuiFab-root"
        style="margin: auto; display: flex;"
        tabindex="0"
        type="submit"
      >
        x
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</DocumentFragment>
`;
